include/libmpu/mpu6050.hpp

```diff
@@ -1,5 +1,201 @@
+    // Bit 7 | Bit 6 | Bit 5 | Bit 4 | Bit 3 | Bit 2 | Bit 1 | Bit 0
+    //  XD15 | XD14  |  XD13 |  XD12 |  XD11 |  XD10 |  XD9  |  XD8
+    //
+    // Final X-axis Byte (LSB)
+    // =========================================================================
+    // Bit 7 | Bit 6 | Bit 5 | Bit 4 | Bit 3 | Bit 2 | Bit 1 | Bit 0
+    //   XD7 |   XD6 |   XD5 |   XD4 |   XD3 |   XD2 |   XD1 |   XD0
+    //
+
+    // We simply shift and OR the bytes together to get them into a signed int
+    // 16 value.
+    int16_t x =
+        static_cast<int16_t>(xyz_acceleration[0] << 8 | xyz_acceleration[1]);
+    int16_t y =
+        static_cast<int16_t>(xyz_acceleration[2] << 8 | xyz_acceleration[3]);
+    int16_t z =
+        static_cast<int16_t>(xyz_acceleration[4] << 8 | xyz_acceleration[5]);
+
+    // Convert the 16 bit value into a floating point value m/S^2
+    constexpr float kMax =
+        static_cast<float>(std::numeric_limits<int16_t>::max());
+    constexpr float kMin =
+        static_cast<float>(std::numeric_limits<int16_t>::min());
+
+    auto int16_bounds  = std::make_pair(kMax, kMin);
+    auto posneg_bounds = std::make_pair(-m_gscale, m_gscale);
+
+    acceleration.x = hal::map(x, int16_bounds, posneg_bounds);
+    acceleration.y = hal::map(y, int16_bounds, posneg_bounds);
+    acceleration.z = hal::map(z, int16_bounds, posneg_bounds);
+
+    return acceleration;
+  }
+
+  hal::status active_mode(bool is_active = true)
+  {
+    constexpr auto sleep_mask = hal::bit::mask::from<6>();
+
+    auto control =
+        HAL_CHECK(hal::write_then_read<1>(*m_i2c,
+                                          m_address,
+                                          std::array{ initalizing_register },
+                                          hal::never_timeout()));
+
+    hal::bit::modify(control[0]).insert<sleep_mask>(!is_active);
+
+    HAL_CHECK(hal::write(*m_i2c,
+                         m_address,
+                         std::array{ initalizing_register, control[0] },
+                         hal::never_timeout()));
+
+    return hal::success();
+  }
+
+  hal::status is_valid_device()
+  {
+    // Verify that the device is the correct device
+    hal::byte expected_device_id = 0x68;
+    // Read out the identity register
+    auto device_id = HAL_CHECK(hal::write_then_read<1>(
+        *m_i2c, m_address, std::array{ who_am_i }, hal::never_timeout()));
+
+    if (device_id[0] != expected_device_id)
+    {
+      return hal::new_error(std::errc::illegal_byte_sequence);
+    }
+
+    return hal::success();
+  }
+
+  hal::status power_on()
+  {
+    HAL_CHECK(is_valid_device());
+    return active_mode(true);
+  }
+
+  hal::status power_off()
+  {
+    return active_mode(false);
+  }
+
+  /// The I2C peripheral used for communication with the device.
+  hal::i2c * m_i2c;
+  /// The configurable device address used for communication.
+  hal::byte m_address;
+
+  float m_gscale = 2.0f;
```

Comment:
```suggestion
  float m_gscale = 2.0f;
  hal::byte m_address;
```
We arrange these search that the packing of the object is more efficient.